frontend/src/components/Card.js

```diff
@@ -1,20 +1,33 @@
 import React from 'react';
 import Rating from './Rating';
 import { Link } from 'react-router-dom';
-import { ADD_TO_WISHLIST } from '../constants/WishListConstants';
+import { ADD_TO_WISHLIST } from '../constants/type';
 import useCustomContext from '../customHooks/Hook';
+import axios from 'axios';
+import Config from '../config/Config';
+import { auth } from '../utlis/auth';
+import { toast } from 'react-toastify';
+
+const { serverUrl } = Config;
 
 const Card = ({ product }) => {
 
+    const token = auth();
     const { state: { wishlist }, dispatch } = useCustomContext();
-    const { id, name, image, price, numReviews, rating } = product;
+    const { _id, name, image, price, numReviews, rating } = product;
 
-    const addToWishlist = () => {
-        dispatch({ type: ADD_TO_WISHLIST, payload: product })
+    const addToWishlist = async () => {
+        try {
+            const { data } = await axios.post(`${serverUrl}/api/wishlists`, { 'productId': _id }, { headers: token });
+            dispatch({ type: ADD_TO_WISHLIST, payload: data.productsinWishlist })
+        } catch (err) {
+            const error = err.response.data.message;
+            toast.error(`${error}`);
+        }
     }
 
     const check = () => {
-        return !!wishlist.find((x) => x.id === id);
+        return !!wishlist.find((x) => x.productId._id === _id);
```

Comment:
If you only want to return a Boolean value, wishlist.some() can be used instead.

Reply:
That's great
done